tools/embed_c.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+import sys
+
+
+def main():
+    import argparse
+    parser = argparse.ArgumentParser(description='Create c/h file from a resource to embed inside a program.')
+    parser.add_argument('input', nargs=argparse.OPTIONAL, default=sys.stdin, type=argparse.FileType('rb'),
+                        help='Input file to embed (default is stdin')
+    parser.add_argument('--output', '-o', type=argparse.FileType('w'), default=sys.stdout,
+                        help='Output C source file (default is stdout)')
+    parser.add_argument('--header', '-p', default=None,
+                        help='Output H header file  (default is to not generate it)')
+    parser.add_argument('--name', '-n', nargs=argparse.OPTIONAL, default='data',
+                        help='Variable name of the generated variable')
+
+    if not sys.version_info >= (3, ):
+        parser.error('This script requires python 3+')
+
+    ns = parser.parse_args()
+
+    # Count the number of bytes while we are reading.
+    # We can't do seek because the input can be stdin which is not seekable.
+    data_length = 0
+
+    # Get the buffer of the stream to read binary data
+    inputfile = ns.input
+    if hasattr(inputfile, 'buffer'):
+        inputfile = inputfile.buffer
+
+    # Write the C file
+    cout = ns.output
+    if ns.header:
+        cout.write('#include "{}"\n'.format(ns.header))
+        cout.write('\n')
+    cout.write('char {}[] = {{\n'.format(ns.name))
+    while True:
+        line = inputfile.read(8)
+        data_length += len(line)
+        if not line:
+            break
+        cout.write('    {}\n'.format(' '.join('0x{:02x},'.format(b) for b in line)))
```

Comment:
Replace the four spaces with a single `\t`.

Reply:
Script has been redownloaded from the gist.